### ISSUE TITLE
Unable to Use in Enterprise due to Struts Dependency

### ISSUE THREAD

**Issue by jonathanamartin** [mojohaus/versions-maven-plugin#204]
Due to the [Struts vulnerability](https://struts.apache.org/docs/s2-052.html), Struts has been completely block at my company.  We had used the `versions-maven-plugin`, however, are no longer able to use it due to the Struts transitive dependency found in Doxia.

```
[INFO] |  +- org.apache.velocity:velocity-tools:jar:2.0:compile
    [INFO] |  |  +- commons-beanutils:commons-beanutils:jar:1.7.0:compile
    [INFO] |  |  +- commons-digester:commons-digester:jar:1.8:compile
    [INFO] |  |  +- commons-chain:commons-chain:jar:1.1:compile
    [INFO] |  |  +- commons-validator:commons-validator:jar:1.3.1:compile
    [INFO] |  |  +- dom4j:dom4j:jar:1.1:compile
    [INFO] |  |  +- oro:oro:jar:2.0.8:compile
    [INFO] |  |  +- sslext:sslext:jar:1.2-0:compile
    [INFO] |  |  +- org.apache.struts:struts-core:jar:1.3.8:compile**
    [INFO] |  |  |  \- antlr:antlr:jar:2.7.2:compile
    [INFO] |  |  +- org.apache.struts:struts-taglib:jar:1.3.8:compile
    [INFO] |  |  \- org.apache.struts:struts-tiles:jar:1.3.8:compile**
```

Could we move the Doxia dependency to a Maven build profile so it's not resolved when just using the plugin?

**Comment by khmarbaise** [comment]
If you are using the plugin it contains the dependencies cause the plugin offers reporting functionality which means the velocity/doxia part is necessary. Furthermore moving things into a profile will not solve the issue at all (if we assume there exists an issue) nor does it help in this case cause than those dependencies are not available at Run Time of the plugin which in consequence means the whole functionality of reporting goals would not be available at all...

furthermore is the question: If you have used versions-maven-plugin before those artifacts have already been in your repository (manager I hope you are using one) inside your network for a long time...based on the first usage of that plugin...

Furthermore if you read your given link: 
```
A RCE attack is possible when using the Struts REST plugin with XStream handler to deserialise XML requests
```
I don't see any hint about the Struts REST plugin in the dependency list? Do you?
Another thing is that based on the [docs the Struts REST Plugin is available with version 2.1.1+ of struts](http://struts.apache.org/docs/rest-plugin.html). In the dependency list we are using struts-1.X no rest plugin there? Or do I miss a thing?

Apart from that If we would like to get struts out there we needed to start at velocity using other versions of velocity or completely replace velocity in doxia...and then we can upgrade in versions-maven-plugin...which will take a longer time...If you like you can join and help to get that done ...Maven/Doxia/Versions-Maven Plugin are open source and help is always appreciated...

**Comment by jonathanamartin** [comment]
Thanks for the quick response.  

> If you are using the plugin it contains the dependencies cause the plugin offers reporting functionality which means the velocity/doxia part is necessary.

We're just using the [`set` goal](http://www.mojohaus.org/versions-maven-plugin/set-mojo.html) to update the version so no reporting is used currently.  

Regarding the Maven profile I was thinking you'd essentially end up with an additional qualified version `-reporting`.  However, I see your point, it's nice to have it all in one place and use functionality based on goals.

> In the dependency list we are using struts-1.X no rest plugin there? Or do I miss a thing?

I agree that the version used in the plugin is older than the one that had the vulnerability.  It was previously stored on an internal Maven repository.  However, our security folks have purged all Struts dependencies.   I don't have control over that.

> Apart from that If we would like to get struts out there we needed to start at velocity using other versions of velocity or completely replace velocity in doxia...and then we can upgrade in versions-maven-plugin...which will take a longer time...If you like you can join and help to get that done ...Maven/Doxia/Versions-Maven Plugin are open source and help is always appreciated...

Agreed, start with the source.  Also, would love to contribute.  Will do when time permits.  However, I need something a little sooner.   For now, I'll just take a fork and exclude the Doxia dependency.
